### ISSUE TITLE
[BUG] Not enough gas supplied when adding fourth TrustedAlgorithm to service

### ISSUE THREAD

**Issue by PixxelFD** [deltaDAO/nautilus#141]
## Summary
When editing a service and trying to add a fourth trusted algorithm to a compute dataset an error occurs.

## Current Behavior

When trying to add a trusted algorithm to a service of a compute asset the following error occurs if already three algorithms have been defined:
`Error: processing response error (body="{\"jsonrpc\":\"2.0\",\"id\":550,\"error\":{\"code\":-32600,\"message\":\"not enough gas supplied for intrinsic gas costs\"}}", error={"code":-32600}, requestBody="{\"method\":\"eth_estimateGas\",\"params\":[{\"from\":\"0x4a806a4851472f7cfd579d3ff5465f03c3c2b5d4\",\"to\":\"0xd275d24f99c102d6addf165df60f2d99df7fcfa0\",\"data\":\"0x1aa3adf900000[...]000000000\"}],\"id\":550,\"jsonrpc\":\"2.0\"}", requestMethod="POST", url="https://rpc.genx.minimal-gaia-x.eu", code=SERVER_ERROR, version=web/5.7.1)`
  
## Expected Behavior
The fourth algorithm should be added without error.

## Steps to Reproduce

1. With the config: 
   `chainId: 100,
    network: 'genx',
    metadataCacheUri: 'https://aquarius510.v4.delta-dao.com',
    nodeUri: 'https://rpc.genx.minimal-gaia-x.eu',
    providerUri: 'https://provider.v4.genx.delta-dao.com',
    subgraphUri: 'https://subgraph.v4.genx.minimal-gaia-x.eu',
    oceanTokenAddress: '0x0995527d3473b3a98c471f1ed8787acd77fbf009',
    oceanTokenSymbol: 'OCEAN',
    fixedRateExchangeAddress: '0xAD8E7d2aFf5F5ae7c2645a52110851914eE6664b',
    dispenserAddress: '0x94cb8FC8719Ed09bE3D9c696d2037EA95ef68d3e',
    nftFactoryAddress: '0x6cb85858183B82154921f68b434299EC4281da53',
    providerAddress: '0x68C24FA5b2319C81b34f248d1f928601D2E5246B',`
2. Use addTrustedAlgorithms function and call nautilus.edit()
3. Tested with the following EuPG-algorithms: 
   `did:op:a2fd44a065902b9b84fc816e500dbf3cdcaf2a31a5b9cabf167c90025b896cf0`
   `did:op:6a6172d10f41fe25529b9fb8204952890cb702040b1bbe2b6e58667bce30ac42`
   `did:op:bd74d6a281ba414de2b4d8ee4087277575f95676bd74e20ee9e2960c9c38d7c5`
   `did:op:a3da777fd3711da36d5e1e5904a8c074b6e8df51549db2b6c8a5bc7ec3ab60cf`

## Environment
- nautilus 1.0.2
- npm 9.8.1

  
## Anything else

@moritzkirstein

**Comment by moritzkirstein** [comment]
Thanks for submitting the issue. The error looks like the account used to edit the assets might not have enough tokens to pay the required gas fees. Can you confirm that the wallet used to edit was / is sufficiently funded with enough gas for the transaction?

We'll have a look into the issue within our current sprint and update here once we have further insights.

**Comment by PixxelFD** [comment]
Yes, i can confirm that the wallet should be sufficiently funded to pay the gas fee (>6 GX) and in addition all other transactions like for publishing still work without issue.
Thanks for looking into it.
